synapse/lib/storm.py

```diff
@@ -5736,6 +5738,10 @@ async def execStormCmd(self, runt, genr):
             opts = {'vars': path.vars}
 
             async with await core.snap(user=user, view=runt.snap.view) as snap:
+                if self.opts.show_msgs:
+                    snap.on('warn', runt.snap.dist)
```

Comment:
is `snap` still gc'd even though its holding a reference to `runt.snap.dist` ?

Reply:
yeah, wouldn't it get cleaned up when `runt` gets destroyed?

Reply:
If we were relying on `runt` being destroyed we could have an unbound number of nodes running through and causing extra memory pressure. We'd need to call `off()` or use `onWith` context managers.